v1/README_Cloud_PubSub_to_Datadog.md

```diff
@@ -0,0 +1,199 @@
+
+### Templates Plugin
+
+This README provides instructions using
+the [Templates Plugin](https://github.com/GoogleCloudPlatform/DataflowTemplates#templates-plugin)
+. Install the plugin with the following command before proceeding:
+
+```shell
+mvn clean install -pl plugins/templates-maven-plugin -am
+```
+
+### Building Template
+
+This template is a Classic Template, meaning that the pipeline code will be
+executed only once and the pipeline will be saved to Google Cloud Storage for
+further reuse. Please check [Creating classic Dataflow templates](https://cloud.google.com/dataflow/docs/guides/templates/creating-templates)
+and [Running classic templates](https://cloud.google.com/dataflow/docs/guides/templates/running-templates)
+for more information.
+
+#### Staging the Template
+
+If the plan is to just stage the template (i.e., make it available to use) by
+the `gcloud` command or Dataflow "Create job from template" UI,
+the `-PtemplatesStage` profile should be used:
+
+```shell
+export PROJECT=<my-project>
+export BUCKET_NAME=<bucket-name>
+
+mvn clean package -PtemplatesStage  \
+-DskipTests \
+-DprojectId="$PROJECT" \
+-DbucketName="$BUCKET_NAME" \
+-DstagePrefix="templates" \
+-DtemplateName="Cloud_PubSub_to_Datadog" \
+-pl v1 \
+-am
+```
+
+The command should build and save the template to Google Cloud, and then print
+the complete location on Cloud Storage:
+
+```
+Classic Template was staged! gs://<bucket-name>/templates/Cloud_PubSub_to_Datadog
+```
+
+The specific path should be copied as it will be used in the following steps.
+
+#### Running the Template
+
+**Using the staged template**:
+
+You can use the path above run the template (or share with others for execution).
+
+To start a job with the template at any time using `gcloud`, you are going to
+need valid resources for the required parameters.
+
+Provided that, the following command line can be used:
+
+```shell
+export PROJECT=<my-project>
+export BUCKET_NAME=<bucket-name>
+export REGION=us-central1
+export TEMPLATE_SPEC_GCSPATH="gs://$BUCKET_NAME/templates/Cloud_PubSub_to_Datadog"
+
+### Required
+export INPUT_SUBSCRIPTION=<inputSubscription>
+export SITE=<site>
+export OUTPUT_DEADLETTER_TOPIC=<outputDeadletterTopic>
+
+### Optional
+export API_KEY=<apiKey>
+export BATCH_COUNT=<batchCount>
+export PARALLELISM=<parallelism>
+export INCLUDE_PUBSUB_MESSAGE=<includePubsubMessage>
+export API_KEY_KMSENCRYPTION_KEY=<apiKeyKMSEncryptionKey>
+export API_KEY_SECRET_ID=<apiKeySecretId>
+export API_KEY_SOURCE=<apiKeySource>
+export JAVASCRIPT_TEXT_TRANSFORM_GCS_PATH=<javascriptTextTransformGcsPath>
+export JAVASCRIPT_TEXT_TRANSFORM_FUNCTION_NAME=<javascriptTextTransformFunctionName>
+
+gcloud dataflow jobs run "cloud-pubsub-to-datadog-job" \
+  --project "$PROJECT" \
+  --region "$REGION" \
```

Comment:
In our testing, we also used `[--enableStreamingEngine](https://cloud.google.com/dataflow/docs/streaming-engine)` to enable the Streaming Engine, in an attempt to minimize costs for customers and enable automatic horizontal scaling. Is this a good recommendation?

Reply:
Also, am I write in saying we need to add roles to the [default Compute Engine service account](https://cloud.google.com/compute/docs/access/service-accounts#default_service_account) everywhere? Is that only bc we used the Streaming Engine? I ask bc it seems there is also a [default Dataflow service account](https://cloud.google.com/dataflow/docs/concepts/security-and-permissions#df-service-account)

Reply:
Agreed, Streaming Engine is a very good idea and very cost worthy whenever reading from Pub/Sub. This documentation can be generated and overridden, so we may need to think on doing something in the [template](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/plugins/core-plugin/src/main/resources/README-template.md)